vncviewer/i18n.h

```diff
@@ -24,10 +24,10 @@
 
 /* Need to tell gcc that pgettext() doesn't screw up format strings */
 #ifdef __GNUC__
-static const char *
+__attribute__ ((format_arg (3))) static const char *
 pgettext_aux (const char *domain,
               const char *msg_ctxt_id, const char *msgid,
-              int category) __attribute__ ((format_arg (3)));
+              int category);
```

Comment:
I don't quite understand this one. This code should be for gcc only. Does clang also declare `__GNUC__`?

And why does it only complain about this one? We have similar things in `Exception.h` and `LogWriter.h`.

Reply:
Yes, clang declares` __GNUC__`. The attribute usage in `i18n.h`/`gettext.h` is different than the two files you reference in that `i18n.h` includes `gettext.h`, which defines `pgettext_aux`, causing a conflict.

Reply:
Ah, I see. I think these files come from upstream gettext. Have you checked if they have similar fixes? It's easier if we copy what they do in case we want to update things.